src/components/GestureButtons.tsx

```diff
@@ -92,6 +92,11 @@ export interface BaseButtonProps extends RawButtonProps {
    * Defaults to 600.
    */
   delayLongPress?: number;
+
+  /**
+   * Used to pass ref from parent into `BaseButton`
+   */
+  innerRef?: React.ForwardedRef<unknown>;
```

Comment:
Also, you mentioned not wanting `innerRef` to be accessible, I think we are exporting this type publicly.

Reply:
Yup, I've missed that. I've changed that in [cf193d1](https://github.com/software-mansion/react-native-gesture-handler/pull/2903/commits/cf193d1a528060fed55418c9c06b8fdac20cb567), though I'm not sure if this is the best way to do this. Maybe there are some TypeScript tricks that we could use to make it look better, cc @tjzel

Reply:
I can't really think of a solution here that wouldn't need any extensive changes sadly :c